os/mm/mm_heap/mm_node.h

```diff
@@ -24,14 +24,48 @@
  ****************************************************************************/
 
 #include <assert.h>
+#include <debug.h>
+
+#include <tinyara/mm/mm.h>
 
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 
+#if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
+#define DEBUGASSERT_MM_FREE_NODE(heap, node)		\
```

Comment:
Do you have any reasons to separate DEBUGASSERT_MM_NODE according to build type? I think both are same except setting of `abort_mode`. If it is not necessary as I said in below comment, we don't need to separate them.